modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManager.java

```diff
@@ -773,24 +850,76 @@ public SCIMResponse updateWithPATCH(String existingId, String scimObjectString,
     /*
      * Creates the Listed Resource.
      *
-     * @param users
-     * @param startIndex
-     * @param totalResults
+     * @param users         List of users retrieved from the user store
+     * @param startIndex    Offset
+     * @param totalResults  Total results retrieved
+     * @param direction     The direction of
      * @return
      * @throws CharonException
      * @throws NotFoundException
      */
-    protected ListedResource createListedResource(List<Object> users, int startIndex, int totalResults)
+    protected ListedResource createListedResource(List<Object> users, Integer startIndex, String cursor,
+                                                  int totalResults, Integer limit)
             throws CharonException, NotFoundException {
+
         ListedResource listedResource = new ListedResource();
         listedResource.setSchema(SCIMConstants.LISTED_RESOURCE_CORE_SCHEMA_URI);
-        listedResource.setTotalResults(totalResults);
-        listedResource.setStartIndex(startIndex);
         listedResource.setItemsPerPage(users.size());
+        listedResource.setTotalResults(totalResults);
+        if (cursor == null) { //set startIndex if its an offset query
+            listedResource.setStartIndex(startIndex);
+        }
         for (Object user : users) {
             Map<String, Attribute> userAttributes = ((User) user).getAttributeList();
             listedResource.setResources(userAttributes);
         }
+        int nextCursorLoc = 0;
+        if ((cursor != null) && (limit != null)) {
+
```

Comment:
Remove empty lines in these places. Add empty lines in the methods if there is a logical separation or if you think it is required to separate the logic to make it readable

Reply:
ack